crates/sim2h/src/lib.rs

```diff
@@ -503,14 +507,14 @@ impl Sim2h {
                     (agents_in_space, aspects_missing_at_node)
                 };
 
-                let aspect_hashes = aspects_missing_at_node.aspect_hashes();
-                if aspect_hashes.len() > 0 {
+                let missing_hashes: HashSet<(EntryHash, AspectHash)> = (&aspects_missing_at_node).into();
+                if missing_hashes.len() > 0 {
                     let mut space = self.spaces
                         .get(space_address)
                         .expect("This function should not get called if we don't have this space")
                         .write();
-                    for aspect_hash in aspect_hashes.clone() {
-                        space.add_missing_aspect(agent_id.clone(), aspect_hash);
+                    for pair in missing_hashes {
```

Comment:
You could de-structure here to make it more readable e.g.
```
for (entry_hash, aspect_hash) in missing_hashes
```
and this avoids having to use `pair.0, pair.1`